onnxruntime/core/graph/function_utils.cc

```diff
@@ -332,17 +333,16 @@ std::unique_ptr<ONNX_NAMESPACE::OpSchema> CreateSchema(const std::string& functi
     }
   }
 
+  // model_local_functions is a member of Model instance and will be alive at the time this is invoked.
   op_schema->TypeAndShapeInferenceFunction(
-      [onnx_func_proto, func_domain_to_version, &model_local_functions](ONNX_NAMESPACE::InferenceContext& ctx) {
+      [onnx_func_proto, func_domain_to_version = std::move(func_domain_to_version), &model_local_functions](ONNX_NAMESPACE::InferenceContext& ctx) {
         auto schema_registry = ONNX_NAMESPACE::OpSchemaRegistry::Instance();
 
         constexpr bool check_type_true = true;
         constexpr int error_mode_throw = 1;
         constexpr bool enable_data_propagation_false = false;
         ONNX_NAMESPACE::ShapeInferenceOptions options{check_type_true, error_mode_throw, enable_data_propagation_false};
 
-        std::unordered_map<std::string, const ONNX_NAMESPACE::FunctionProto*> map_copy(model_local_functions.begin(),
-                                                                                       model_local_functions.end());
         std::unordered_map<std::string, TensorShapeProto> empty_map;
```

Comment:
>std::unordered_map<std::string, TensorShapeProto> empty_map; [](http://example.com/codeflow?start=8&length=60)

Can we pass nullptr instead of this throw away map?